python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -697,39 +889,164 @@ def setDefaults(self):
-
-    This is done in a number of steps:
-        - identify potential regions that need to be clipped
-        - find how much overlap there is between each potential region and the DETECTED mask for each visit
-        - apply a cut based on the visit overlaps to determine the clipped regions
-        - for big areas do not use the region as determined from the coadd, but rather all overlapping
-          footprints from the visit DETECTED plane.  This helps in flagging low surface brightness regions.
-        - assemble the coadd, ignoring the clipped regions.
-
-    The cuts used to determine the clipped regions were found empirically by examining a lot of HSC data.
-    It is unlikely that they will generalize well to other data sets.  The current algorithm is limited to
-    identifying artifacts that are found in only one or two visits.  A different threshold is applied
-    depending on the number overlapping visits.  The current thresholds work well at removing a majority
-    of artifacts.
-
+    """!
+    \anchor SafeClipAssembleCoaddTask
+
+    \brief Assemble a coadded image from a set of coadded temporary exposures, being careful to clip & flag areas 
+    with potential artifacts.
+
+    \section pipe_tasks_assembleCoadd_Contents Contents
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Purpose
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Initialize
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Run
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Config
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Debug
+      - \ref pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Example
+
+    \section pipe_tasks_assembleCoadd_SafeClipAssembleCoaddTask_Purpose	Description
+
+    \copybrief SafeClipAssembleCoaddTask
+
+    We want to assemble a coadded image from a set of coadded temporary exposures (coaddTempExps).
+    We implicitly assume that the coaddTempExps were all observed in the same filter band.
+    Each input coaddTempExp covers a patch on the sky and corresponds to a single run/visit/exposure of the
+    covered patch.
+    We provide the task with a list of coaddTempExps (selectDataList) from which it selects coaddTempExps that
+    cover the specified patch (pointed at by dataRef).
+    Each coaddTempExp that goes into a coadd will typically have an independent photometric zero-point.
+    Therefore, we must scale each coaddTempExp to set it to a common photometric zeropoint.
+    By default, each coaddTempExp has backgrounds and hence will require config.doMatchBackgrounds=True.
+    When background matching is enabled, the task may be configured to automatically select a reference
+    exposure (config.autoReference=True).
+    If this is not done, we require that the input dataRef provides access to a coaddTempExp (dataset type
+    coaddName + 'Coadd_tempExp') which is used as the reference exposure.
```

Comment:
I don't think you need to repeat the same documentation that is in AssembleCoaddTask.